### PR TITLE
Add CRFCut sentence segmentation

### PR DESCRIPTION
**Pull request by cstorm125**
CRFCut -- Thai sentence segmentation with conditional random field, default trained on TED dataset

- ORCHID - space-correct accuracy 87% vs 95% state-of-the-art (Zhou et al, 2016; https://www.aclweb.org/anthology/C16-1031.pdf)
- TED dataset - space-correct accuracy 82%

See development notebooks at https://github.com/vistec-AI/ted_crawler;
POS features are not used due to unreliable POS tagging available

### CODE REVIEW

**Comment by bact** [comment]
1) `python-crfsuite` should be included in `setup.py`'s `install_requires` (and maybe in `requirements.txt` as well, see next point)
  - The code currently passed the test only because in the test the `python-crfsuite` was installed as dependencies of `attacut` - but in real settings, user may not install the optional `attacut`.

2) How big is the model? If it's not that big I would suggest to include it in the module, so we will less dependent on the network during runtime.
  - This will make it possible to have `crfcut` included as a pythainlp core, as a default engine for sentence segmentation.
  - If this is really the case, `python-crfsuite` should be in `requirements.txt` too.
  - The model file should be included in `setup.py`'s `package_data`
  - Once the model included, `_download()` function should be removed.

**Comment by cstorm125** [comment]
@bact The model is 5MB so I agree we can include it as battery. Is everyone okay with having some model files in the library? @artificiala @wannaphongcom

**Comment by bact** [comment]
For reference on model size, see https://github.com/PyThaiNLP/pythainlp/issues/298 

I think 5 MB is ok.

File: pythainlp/tokenize/__init__.py [hunk 1]
```diff
@@ -179,17 +179,17 @@ def dict_word_tokenize(
...
+    This function tokenizes running text into "sentences"
 
     :param str text: the text to be tokenized
-    :param str engine: choose between *'whitespace'* or *'whitespace+newline'*
+    :param str engine: choose among *'crfcut'*, *'whitespace'*, \
+    *'whitespace+newline'*
     :return: list of splited sentences
     :rtype: list[str]
     **Options for engine**
+        * *crfcut* - split by CRF trained on TED dataset
```
Comment (wannaphong) [inline pythainlp/tokenize/__init__.py]: `* *crfcut* - split by CRF trained on TED dataset` -> `* *crfcut* (default) - split by CRF trained on TED dataset`

File: pythainlp/tokenize/__init__.py [hunk 2]
```diff
@@ -179,17 +179,17 @@ def dict_word_tokenize(
...
         * *whitespace+newline* (default) - split by whitespace token \
```
Comment (wannaphong) [inline pythainlp/tokenize/__init__.py]: `* *whitespace+newline* (default) - split by whitespace token` -> `* *whitespace+newline* - split by whitespace token`

**Review by bact — CHANGES_REQUESTED** [review]
I have add few words to STARTERS and ENDERS. May require retrain and reupload of the model.
Apart of that, I think we're good to go.

Please also kindly update the table here: https://github.com/PyThaiNLP/pythainlp/issues/298 thx

Great work! Another step towards full pipeline.

**Comment by wannaphong** [comment]
:100:

**Comment by bact** [comment]
TODO: Next step is to convert https://github.com/vistec-AI/ted_crawler/blob/master/sentenceseg_ted.ipynb to a commandline script and maybe put it in `bin/` directory. So people can train their own model. Follow up on #344.

**Pull request merged**